src/Illuminate/Bus/Queueable.php

```diff
@@ -96,6 +138,10 @@ public function dispatchNextJobInChain()
         if (! empty($this->chained)) {
             dispatch(tap(unserialize(array_shift($this->chained)), function ($next) {
                 $next->chained = $this->chained;
+                $next->chainConnection = $this->chainConnection;
```

Comment:
This does not support having the chain connection change with the next job having the queue/connection already set...
```php
$next->chainConnection = $next->chainConnection??$this->chainConnection;
```

Reply:
Yeah we don't want to do that, the chain connection is declared ones and used as a default connection for when no connection is specified.

Reply:
Interesting. I am not sure why that was not communicated on my pull request. Keeps it simpler, Thanks.